core/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python:3.12 AS lomas_core_base
+
+WORKDIR /code
+
+COPY ./requirements_core.txt /code/requirements_core.txt
+RUN pip install --no-cache-dir --upgrade -r /code/requirements_core.txt
+
+COPY ./lomas_core/ /code/lomas_core/
+COPY ./setup.py /code/setup.py
+COPY ./README.md /code/README.md
+
+FROM lomas_core_base AS lomas_core_dev
+RUN pip install -e /code/
```

Comment:
This idea of the dev images was not to copy the source files but to mount them at container start up, so that if we change code in an IDE, it is reflected in the docker.

Here we can either: 
- Only keep the lomas_core image (It could be that mounting the files at startup will overwrite the ones in the container and still allow the quick development loop)
- Keep the lomas_core_dev  without a pip install -e and move the file copying to the lomas_core one.

Reply:
Not sure I understood why there is a dev image and another one ? 

Would just the start work ? maybe that's why the client does not work in the notebook for now

```
FROM python:3.12 AS lomas_core

WORKDIR /code

COPY ./requirements_core.txt /code/requirements_core.txt
RUN pip install --no-cache-dir --upgrade -r /code/requirements_core.txt

COPY ./lomas_core/ /code/lomas_core/
COPY ./setup.py /code/setup.py
COPY ./README.md /code/README.md

RUN pip install -e /code/
```

Reply:
This one would definitely work for the published image.

For the dev image, I wanted to explicitly NOT copy the source files to the image so that the developper is forced to mount the files at runtime. Like this we make sure the changes she/he applies to the notebooks take effect in his/her own copy of the git repo and not everything is lost when the container is stopped. This is especially useful for the client container, maybe less so for the server one.